src/composables/usePlugins.ts

```diff
@@ -0,0 +1,107 @@
+
+// aggregate all plugin.json files in src/plugins
+const pluginIndex: Record<string, any> = Object.fromEntries(
+  Object.entries(import.meta.globEager('../plugins/*/plugin.json')).map(
+    ([path, config]) => {
+      const pluginKey = path
+        .replace('../plugins/', '')
+        .replace('/plugin.json', '');
+      return [pluginKey, { key: pluginKey, ...config }];
+    }
+  )
+);
+
+// prepare all plugin's main components imports (Create.vue, Proposal.vue, etc.)
+const allPluginComponents = import.meta.glob(`../plugins/*/*.vue`);
+
+// get required components for specific location (componentName) and list of active plugins
+const getPluginComponents = (componentName: string, pluginKeys) => {
+  pluginKeys = pluginKeys.filter(key => !!pluginIndex[key]); // remove old/non-existent plugins
+
+  return Object.entries(allPluginComponents)
+    .filter(([path]) => {
+      if (path.endsWith(componentName + '.vue')) {
+        const pluginKey = path
+          .replace('../plugins/', '')
+          .replace(`/${componentName}.vue`, '');
+        return pluginKeys.includes(pluginKey);
+      }
+      return false;
+    })
+    .map(comp => defineAsyncComponent(comp[1]));
+};
+
+// takes the space's enabled plugins (pluginKeys), checks if they have the
+// specified hook (hookName) in their plugin.json, imports and executes them
+const executePluginHooks = async (hookName, pluginKeys, payload) => {
+  pluginKeys = pluginKeys.filter(key => !!pluginIndex[key]); // remove old/non-existent plugins
+
+  for (let i = 0; i < pluginKeys.length; i++) {
+    const hookPath = pluginIndex[pluginKeys[i]].hooks?.[hookName]
+      ?.replace(/\.ts$/i, '')
+      .replace(/^\.\//, '');
+    if (hookPath) {
+      const { default: hook } = await import(
+        `../plugins/${pluginKeys[i]}/${hookPath}.ts`
+      );
+      await hook(payload);
+    }
+  }
+};
+
+// space count and filter function
+const pluginsSpacesCount: any = ref(null);
+const loadingPluginsSpaceCount = ref(false);
+
+const { apolloQuery } = useApolloQuery();
+const getPluginsSpacesCount = async () => {
+  if (pluginsSpacesCount.value) return; // run only once
+
+  loadingPluginsSpaceCount.value = true;
+  const res = await apolloQuery({ query: PLUGINS_COUNT_QUERY }, 'plugins');
+  // turn [{ id: "myPlugin", spaceCount: 1 }, ...] to { myPlugin: 1, ... }
+  pluginsSpacesCount.value = res.reduce(
+    (obj: any, item: any) => ({ ...obj, [item.id]: item.spacesCount }),
+    {}
+  );
+  loadingPluginsSpaceCount.value = false;
+};
+
+const filterPlugins = (q = '') =>
+  Object.values(pluginIndex)
+    .filter(plugin =>
+      JSON.stringify(plugin).toLowerCase().includes(q.toLowerCase())
+    )
+    .sort(
+      (a, b) =>
+        (pluginsSpacesCount.value?.[b.key] ?? 0) -
+        (pluginsSpacesCount.value?.[a.key] ?? 0)
+    );
+
+/**
+ * Composable
+ *
+ * Does it really make sense to use the composable pattern here? Most of it can
+ * be normal imports.
```

Comment:
Not really. I was hoping we could keep `usePluginsFilter` as it was and move the rest to a helper instead.